Bash version of updateAssemblyInfo script

mthalman opened the pull request:
This defines a Bash version of the existing [updateAssemblyInfo.ps1 script](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev/updateAssemblyInfo.ps1). Since there are already existing Bash versions of the build and test scripts, it made sense to follow that pattern here as well.

This accommodates .NET's source-build requirements which disallow the dependency on PowerShell because it's not buildable from source by Linux distros. The implementation in this script is necessary for .NET's source-build in order to set the assembly versions correctly (see https://github.com/dotnet/source-build/issues/3565).

jennyf19 commented:
@mthalman Thanks for the contribution...could you target dev7 branch instead? Thx.

mthalman committed:
Bash version of updateAssemblyInfo script

mthalman committed:
Remove dependency on xmllint tool which might not exist in environment

mthalman commented:
Converted to a draft. I need to make some more changes so that this will run in Alpine Linux.

mthalman committed:
Replace grep with sed

This allows it work in BusyBox where grep doesn't have support for Perl
regex syntax

jennyf19 commented:
Thanks @mthalman

jennyf19 commented:
Included in [preview4 release](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/releases/tag/7.0.0-preview4)
